docs/shared/tutorials/epic-web-app.md

```diff
@@ -0,0 +1,495 @@
+
+Nx plugins can:
+
+- automatically configure caching for you, including inputs and outputs based on the underlying tooling configuration
+- infer tasks that can be run on a project because of the tooling present
+- provide code generators to help scaffold out projects
+- automatically keep the tooling versions and configuration files up to date
+
+For this tutorial, we'll just focus on the automatic caching configuration.
+
+First, let's delete the `outputs` array from `nx.json` so that we don't override the inferred values from the plugin. Your `nx.json` should look like this:
+
+```json {% fileName="nx.json" %}
+{
+  "targetDefaults": {
+    // ...
+    "test:e2e:run": {
+      "cache": true
+    }
+    // ...
+  }
+}
+```
+
+Now let's add the `@nx/playwright` plugin:
+
+```{% command="npx nx add @nx/playwright" path="~/epic-web-app" %}
+✔ Installing @nx/playwright...
+✔ Initializing @nx/playwright...
+
+ NX   Package @nx/playwright added successfully.
+```
+
+The `nx add` command installs the version of the plugin that matches your repo's Nx version and runs that plugin's initialization script. For `@nx/playwright`, the initialization script registers the plugin in the `plugins` array of `nx.json` and updates any `package.json` scripts that execute Vite related tasks. Open the project details view for the app and look at the `e2e` task.
+
+{% callout title="Find your project name" %}
+When your app was created by the `create-epic-app` script, it added a unique identifier to the name in `package.json`. Use your project name in the command below, instead of `-7d25`.
+{% /callout %}
+
+```shell {% path="~/epic-web-app" %}
+npx nx show project epic-web-app-7d25 --web
+```
+
+{% project-details title="Project Details View" jsonFile="shared/tutorials/epic-web-pdv.json" %}
+{% /project-details %}
+
+If you hover over the settings for the `e2e` task, you can see where those settings come from. The `inputs` and `outputs` are defined by the `@nx/playwright` plugin from the `playwright.config.ts` file.
+
+Now let's change where the results are output to in the `playwright.config.ts` file.
+
+```ts {% fileName="playwright.config.ts" highlightLines=[8] %}
+import { defineConfig, devices } from '@playwright/test';
+import 'dotenv/config';
+
+const PORT = process.env.PORT || '3000';
+
+export default defineConfig({
+  testDir: './tests/e2e',
+  outputDir: './playwright.local.test-results',
+  timeout: 15 * 1000,
+  // ...
+});
+```
+
+Now if you look at the project details view again, you'll see that the `outputs` property for Nx's caching has been updated to stay in sync with the setting in the `playwright.config.ts` file.
+
+You can also add the `@nx/remix`, `@nx/vite` and `@nx/eslint` plugins to see how they infer their respective tasks based on the tooling configuration files.
```

Comment:
I wonder whether we should bring them in. I don't know what the story is about crystalizing `@nx/remix`. That way the cache would use the vite config etc.